src/odemis/gui/comp/overlay/cryo_feature.py

```diff
@@ -267,3 +324,27 @@ def set_icon(feature_icon):
                 self._label.draw(ctx)
 
             ctx.paint()
+
+    def _view_to_stage_pos(self, v_pos):
+        """Convert view position to stage position"""
+        p_pos = self.cnvs.view_to_phys(v_pos, self.cnvs.get_half_buffer_size())
+        new_pos = {
+            "x": p_pos[0],
+            "y": p_pos[1],
+            "z": self.tab_data.main.stage.position.value["z"],  #NOTE: we cannot use cnvs.view._stage, because the acquired cnvs does not have a _stage....? why?
+        }
+        pos = self.pm.from_sample_stage_to_stage_position(new_pos)
+        return pos
+
+    def _get_feature_position_at_view_posture(self, feature: CryoFeature) -> Dict[str, float]:
+        """Get the feature position at the view posture, create it if it doesn't exist"""
+
+        return get_feature_position_at_posture(
+            pm=self.pm,
+            feature=feature,
+            posture=self.pm.current_posture.value, #self.view_posture,
```

Comment:
If you call `get_feature_position_at_posture()`, and the position for the current posture doesn't exist, how it is going to work? It seems it'll call `to_posture()` with the current posture, which will return the position as-is (as current posture is the same as the requested posture).

Reply:
the posture positions always exist, they are computed on init: https://github.com/patrickcleeve2/odemis/blob/99ceb86ead44117d1d38342a1422d241bb3401a9/src/odemis/gui/model/tab_gui_data.py#L256